smartsim/_core/generation/generator.py

```diff
@@ -28,206 +28,276 @@
 import os
 import pathlib
 import pickle
-import shutil
 import subprocess
 import sys
 import typing as t
+from collections import namedtuple
 from datetime import datetime
-from os import mkdir, path
-from os.path import join
 
-from ...entity import Application, TaggedFilesHierarchy
 from ...entity.files import EntityFiles
 from ...launchable import Job
 from ...log import get_logger
+from ..commands import Command, CommandList
 
 logger = get_logger(__name__)
 logger.propagate = False
 
 
+@t.runtime_checkable
+class _GenerableProtocol(t.Protocol):
+    """Ensures functions using job.entity continue if attrs file and params are supported."""
+
+    files: t.Union[EntityFiles, None]
+    file_parameters: t.Mapping[str, str]
+
+
+Job_Path = namedtuple("Job_Path", ["run_path", "out_path", "err_path"])
+"""Paths related to the Jobs execution."""
+
+
 class Generator:
-    """The primary job of the Generator is to create the directory and file structure
-    for a SmartSim Job. The Generator is also responsible for writing and configuring
-    files into the Job directory.
-    """
+    """The primary responsibility of the Generator class is to create the directory structure
+    for a SmartSim Job and to build and execute file operation commands."""
+
+    run = "run"
```

Comment:
Would ``run_directory`` be a better variable name?

Reply:
ah yes, run is vague